text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,758 @@
+  - The requirement to provide "direct value" means that it does not suffice to
+    argue that having host tools will help the target's maintainers more easily
+    provide the target to others. The tools themselves must provide value to
+    others.
+- There must be a reasonable expectation that the host tools will be used, for
+  purposes other than to prove that they can be used.
+- The host tools must build and run reliably in CI, though they may or may not
+  pass tests.
+- Building host tools for the target must not take substantially longer than
+  building host tools for other targets, and should not substantially raise the
+  maintenance burden of the CI infrastructure.
+- The host tools must provide a substantively similar experience as on other
+  targets, subject to reasonable target limitations.
+  - Adding a substantively different interface to an existing tool, or a
+    target-specific interface to the functionality of an existing tool,
+    requires design and implementation approval (e.g. RFC/MCP) from the
+    appropriate approving teams for that tool.
+    - Such an interface should have a design that could potentially work for
+      other targets with similar properties.
+    - This should happen separately from the review and approval of the target,
+      to simplify the target review and approval processes, and to simplify the
+      review and approval processes for the proposed new interface.
+  - By way of example, a target that runs within a sandbox may need to modify
+    the handling of files, tool invocation, and similar to meet the
+    expectations and conventions of the sandbox, but must not introduce a
+    separate "sandboxed compilation" interface separate from the CLI interface
+    without going through the normal approval process for such an interface.
+    Such an interface should take into account potential other targets with
+    similar sandboxes.
+- If the host tools for the platform would normally be expected to be signed or
+  equivalent (e.g. if running unsigned binaries or similar involves a
+  "developer mode" or an additional prompt), it must be possible for the Rust
+  project's automated builds to apply the appropriate signature process,
+  without any manual intervention by either Rust developers, target
+  maintainers, or a third party. This process must meet the approval of the
+  infrastructure team.
+  - This process may require one-time or semi-regular manual steps by the
+    infrastructure team, such as registration or renewal of a signing key. Any
+    such manual process must meet the approval of the infrastructure team.
+  - This process may require the execution of a legal agreement with the
+    signature provider. Such a legal agreement may be revocable, and may
+    potentially require a nominal fee, but must not be otherwise onerous. Any
+    such legal agreement must meet the approval of the infrastructure team.
```

Comment:
With my Rust Foundation hat on: I think the final approval and execution of legal agreements will now be overseen by the foundation. Infrastructure team may still have an interest in approving these for one reason or another, but I think it is worth clarifying that they should not be executing legal agreements on behalf of the project, which has no legal entity.

Reply:
@tmandry This policy carefully avoids specifying the precise process by which legal agreements will be reviewed and executed, and who will be the executing parties. I would agree in principle that no legal agreements should require individual developers to be parties to them. However, for logistical reasons I can imagine scenarios where (for instance) a service provider's terms of service might require each person with a user account to individually agree, separate from the *organizational* agreement required, and infra folks may need to have an account as part of infrastructure maintenance and coordination. For that and similar logistical reasons, my intention with this requirement is that the infrastructure team has oversight and approval here, in addition to the review/approval process by the parties to the agreement (which I would indeed hope would be the foundation). I want to make sure anyone on the infrastructure team has the opportunity to go "hey, wait a minute, this is a problem".

Reply:
Makes sense. I agree it's a good idea, noting that I think the approval process ought to be lightweight enough not to slow things down substantially. But I don't think that necessarily needs to be spelled out in the RFC.

I think part of the role of the foundation _should_ be to look out for issues like this and point them out proactively, but I absolutely agree that individuals involved should get a chance to review for themselves.

As for my original point, I don't think there's anything wrong in what the wording says, I just don't want approval to be misconstrued as meaning more than it actually does.

> This policy carefully avoids specifying the precise process by which legal agreements will be reviewed and executed, and who will be the executing parties.

Stating this next to the wording around approval can prevent that misconstrual, and hopefully will encourage people to be cautious around these things. (I'm not a lawyer but I'm trying to think of cases where this could be hastily done in a way that creates legal headaches down the line. For example, clicking "I agree" to a terms of service for a project account rather than an individual one.)

Reply:
Thanks for the explanation! I agree that there's value in spelling that out. I've now added a clarification in that regard to make the purpose of such review clear, and to make sure folks don't think they're personally responsible for signing legal agreements for the project.